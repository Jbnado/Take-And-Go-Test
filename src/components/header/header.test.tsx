import { describe, expect, it } from "vitest";
import TestRenderer from "react-test-renderer";
import { Bell, Gear, PersonFill } from "react-bootstrap-icons";
import { Header } from "./index";

describe("Header Component", () => {
  it("should render correctly", () => {
    const tree = TestRenderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the logo", () => {
    const tree = TestRenderer.create(<Header />);
    const logo = tree.root.findByType("img");
    expect(logo.props.alt).toBe("Logo");
  });

  it("renders the notifications icon", () => {
    const tree = TestRenderer.create(<Header />);
    const notificationsIcon = tree.root.findByType(Bell);
    expect(notificationsIcon.props.size).toBe(16);
  });

  it("renders the settings icon", () => {
    const tree = TestRenderer.create(<Header />);
    const notificationsIcon = tree.root.findByType(Gear);
    expect(notificationsIcon.props.size).toBe(16);
  });

  it("renders the user icon", () => {
    const tree = TestRenderer.create(<Header />);
    const button = tree.root.findByType("button");
    const roundedFull = button.findByType("div");
    const userIcon = roundedFull.findByType(PersonFill);

    expect(button.props.className).toBe(
      "p-1 group flex shrink-0 items-center justify-center rounded-full transition text-blue-400 shadow-sm hover:text-blue-500 border-2 border-blue-600"
    );
    expect(userIcon.props.size).toBe(20);
  });
});
