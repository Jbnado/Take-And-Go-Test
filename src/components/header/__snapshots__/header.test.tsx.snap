// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > should render correctly 1`] = `
<header
  aria-label="Page Header"
  className="bg-white shadow py-4 px-5"
>
  <div
    className="flex items-center sm:justify-between gap-4"
  >
    <div
      className="flex gap-4 items-center"
    >
      <img
        alt="Logo"
        className="h-8 w-auto"
        src="/Logo.png"
      />
      <div
        className="sm:flex hidden h-5 w-5 group shrink-0 items-center justify-center rounded-full transition text-blue-600 border-2 border-blue-600"
      >
        <svg
          fill="currentColor"
          height={14}
          viewBox="0 0 16 16"
          width={14}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.707 1.5a1 1 0 0 0-1.414 0L.646 8.146a.5.5 0 0 0 .708.708L2 8.207V13.5A1.5 1.5 0 0 0 3.5 15h9a1.5 1.5 0 0 0 1.5-1.5V8.207l.646.647a.5.5 0 0 0 .708-.708L13 5.793V2.5a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5v1.293L8.707 1.5ZM13 7.207V13.5a.5.5 0 0 1-.5.5h-9a.5.5 0 0 1-.5-.5V7.207l5-5 5 5Z"
          />
        </svg>
      </div>
      <div
        className="sm:flex hidden gap-2 items-center cursor-pointer"
      >
        <span
          className="font-medium"
        >
          Dryve - Ribeirão Preto
        </span>
        <svg
          fill="currentColor"
          height={12}
          viewBox="0 0 16 16"
          width={12}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="flex sm:gap-7 gap-5 items-center"
    >
      <a
        className="block shrink-0 text-gray-600 hover:text-gray-700"
        href="#"
      >
        <span
          className="sr-only"
        >
          Notificações
        </span>
        <svg
          fill="currentColor"
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 16a2 2 0 0 0 2-2H6a2 2 0 0 0 2 2zM8 1.918l-.797.161A4.002 4.002 0 0 0 4 6c0 .628-.134 2.197-.459 3.742-.16.767-.376 1.566-.663 2.258h10.244c-.287-.692-.502-1.49-.663-2.258C12.134 8.197 12 6.628 12 6a4.002 4.002 0 0 0-3.203-3.92L8 1.917zM14.22 12c.223.447.481.801.78 1H1c.299-.199.557-.553.78-1C2.68 10.2 3 6.88 3 6c0-2.42 1.72-4.44 4.005-4.901a1 1 0 1 1 1.99 0A5.002 5.002 0 0 1 13 6c0 .88.32 4.2 1.22 6z"
          />
        </svg>
      </a>
      <a
        className="block shrink-0 text-gray-600 hover:text-gray-700"
        href="#"
      >
        <span
          className="sr-only"
        >
          Configurações
        </span>
        <svg
          fill="currentColor"
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 4.754a3.246 3.246 0 1 0 0 6.492 3.246 3.246 0 0 0 0-6.492zM5.754 8a2.246 2.246 0 1 1 4.492 0 2.246 2.246 0 0 1-4.492 0z"
          />
          <path
            d="M9.796 1.343c-.527-1.79-3.065-1.79-3.592 0l-.094.319a.873.873 0 0 1-1.255.52l-.292-.16c-1.64-.892-3.433.902-2.54 2.541l.159.292a.873.873 0 0 1-.52 1.255l-.319.094c-1.79.527-1.79 3.065 0 3.592l.319.094a.873.873 0 0 1 .52 1.255l-.16.292c-.892 1.64.901 3.434 2.541 2.54l.292-.159a.873.873 0 0 1 1.255.52l.094.319c.527 1.79 3.065 1.79 3.592 0l.094-.319a.873.873 0 0 1 1.255-.52l.292.16c1.64.893 3.434-.902 2.54-2.541l-.159-.292a.873.873 0 0 1 .52-1.255l.319-.094c1.79-.527 1.79-3.065 0-3.592l-.319-.094a.873.873 0 0 1-.52-1.255l.16-.292c.893-1.64-.902-3.433-2.541-2.54l-.292.159a.873.873 0 0 1-1.255-.52l-.094-.319zm-2.633.283c.246-.835 1.428-.835 1.674 0l.094.319a1.873 1.873 0 0 0 2.693 1.115l.291-.16c.764-.415 1.6.42 1.184 1.185l-.159.292a1.873 1.873 0 0 0 1.116 2.692l.318.094c.835.246.835 1.428 0 1.674l-.319.094a1.873 1.873 0 0 0-1.115 2.693l.16.291c.415.764-.42 1.6-1.185 1.184l-.291-.159a1.873 1.873 0 0 0-2.693 1.116l-.094.318c-.246.835-1.428.835-1.674 0l-.094-.319a1.873 1.873 0 0 0-2.692-1.115l-.292.16c-.764.415-1.6-.42-1.184-1.185l.159-.291A1.873 1.873 0 0 0 1.945 8.93l-.319-.094c-.835-.246-.835-1.428 0-1.674l.319-.094A1.873 1.873 0 0 0 3.06 4.377l-.16-.292c-.415-.764.42-1.6 1.185-1.184l.292.159a1.873 1.873 0 0 0 2.692-1.115l.094-.319z"
          />
        </svg>
      </a>
      <button
        className="p-1 group flex shrink-0 items-center justify-center rounded-full transition text-blue-400 shadow-sm hover:text-blue-500 border-2 border-blue-600"
        type="button"
      >
        <div
          className="rounded-full bg-blue-200 p-1"
        >
          <span
            className="sr-only"
          >
            Usuário
          </span>
          <svg
            fill="currentColor"
            height={20}
            viewBox="0 0 16 16"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3Zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</header>
`;
